tests/stubs/lwipopts.h

```diff
@@ -1,162 +1,66 @@
-  *    software, must execute solely and exclusively on microcontroller or
-  *    microprocessor devices manufactured by or for STMicroelectronics.
-  * 5. Redistribution and use of this software other than as permitted under
-  *    this license is void and will automatically terminate your rights under
-  *    this license.
-  *
-  * THIS SOFTWARE IS PROVIDED BY STMICROELECTRONICS AND CONTRIBUTORS "AS IS"
-  * AND ANY EXPRESS, IMPLIED OR STATUTORY WARRANTIES, INCLUDING, BUT NOT
-  * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
-  * PARTICULAR PURPOSE AND NON-INFRINGEMENT OF THIRD PARTY INTELLECTUAL PROPERTY
-  * RIGHTS ARE DISCLAIMED TO THE FULLEST EXTENT PERMITTED BY LAW. IN NO EVENT
-  * SHALL STMICROELECTRONICS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-  * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-  * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
-  * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-  * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-  * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
-  * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-  *
-  ******************************************************************************
-  */
-
-/* Define to prevent recursive inclusion --------------------------------------*/
 #ifndef __LWIPOPTS__H__
-    #define __LWIPOPTS__H__
-
-/*-----------------------------------------------------------------------------*/
-/* Current version of LwIP supported by CubeMx: 2.0.3 -*/
-/*-----------------------------------------------------------------------------*/
-
-/* Within 'USER CODE' section, code will be kept by default at each generation */
-/* USER CODE BEGIN 0 */
+#define __LWIPOPTS__H__
 
-/* USER CODE END 0 */
-
-    #ifdef __cplusplus
+#ifdef __cplusplus
 extern "C" {
-    #endif
-
-    #include <stdint.h>
-
-    /* STM32CubeMX Specific Parameters (not defined in opt.h) ---------------------*/
-    /* Parameters set in STM32CubeMX LwIP Configuration GUI -*/
-    /*----- WITH_RTOS enabled (Since FREERTOS is set) -----*/
-    #define WITH_RTOS 0
-    /*----- CHECKSUM_BY_HARDWARE disabled -----*/
-    #define CHECKSUM_BY_HARDWARE 0
-    /*-----------------------------------------------------------------------------*/
-
-    /* LwIP Stack Parameters (modified compared to initialization value in opt.h) -*/
-    /* Parameters set in STM32CubeMX LwIP Configuration GUI -*/
-    /*----- Value in opt.h for LWIP_DHCP: 0 -----*/
-    #define LWIP_DHCP 0
-    /*----- Value in opt.h for MEM_ALIGNMENT: 1 -----*/
-    #define MEM_ALIGNMENT 8
-    /*----- Value in opt.h for MEMP_NUM_SYS_TIMEOUT: (LWIP_TCP + IP_REASSEMBLY + LWIP_ARP + (2*LWIP_DHCP) + LWIP_AUTOIP + LWIP_IGMP + LWIP_DNS + (PPP_SUPPORT*6*MEMP_NUM_PPP_PCB) + (LWIP_IPV6 ? (1 + LWIP_IPV6_REASS + LWIP_IPV6_MLD) : 0)) -*/
```

Comment:
I'd probably keep some of the explanatory comments in this case, not sure if they are documented elsewhere

Reply:
They are documented in LwIP in `opt.h` (there are the default values, our config overrides some of them). 

Having the default values here mentioned wasn't bad. But I am sure that nobody will manually document newly introduced config values in the same way, and I did not want to have some of them documented and some not. So I removed the generated docs, as everything can still be found in opt.h...

Reply:
ok